### DISCUSSION TITLE
Weight decay norm

### DISCUSSION THREAD

**Discussion by cemde** [google-deepmind/optax#520] [Q&A]
What does the `optax.add_decayed_weights` function actually implement?

For loss function $L$, penalty coefficient $\lambda$, and layer $l$, does it implement $L + \lambda \sum_l ||W_l||_2$ or  $L +\lambda  \sum_l ||W_l||_1$, i.e. L1 or L2 penalty? Whatever it implements is there a function for the other?
